service/controller/v26/resource/configmap/update.go

```diff
@@ -85,7 +85,8 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 				return nil, microerror.Mask(err)
 			}
 
-			if isConfigMapModified(desiredConfigMap, currentConfigMap) {
+			isModified := !isEmpty(currentConfigMap) && !equals(currentConfigMap, desiredConfigMap)
```

Comment:
the operator didn't compare labels and annotations before. Now we do it.

Reply:
We need to compare labels but didn't we have a problem comparing annotations in `app-operator`?

Reply:
Are you sure we should do this? I cannot tell you from the top of my head if we need to do this or maybe should not.

Reply:
The only resource that using these configmaps is `deployment` resource in the later reconciliation step, and it used by objectName, so it's safe from this resource's point of view.
And I think we should make labels as same as kvm-operator desired.

Reply:
Currently, we are comparing annotation even in app-operator https://github.com/giantswarm/app-operator/blob/703986b1568559b8fcaed5c0bc2d17d964ca2320/service/controller/app/v1/resource/configmap/resource.go#L80-L82
😱 Should we get rid of comparing these annotation values?

Reply:
OK but in the chart resource we had problems which is what I was thinking of.

https://github.com/giantswarm/app-operator/blob/703986b1568559b8fcaed5c0bc2d17d964ca2320/service/controller/app/v1/resource/chart/resource.go#L89-L98

If we are not setting annotations in the desired state why do we need to compare them?

Reply:
@rossf7 You're right, I deleted the part of comparing annotations for these configmaps. PTAL :)